src/lib.rs

```diff
@@ -813,11 +810,11 @@ impl Monitor {
         unsafe {
             let llramp = *ffi::glfwGetGammaRamp(self.ptr);
             GammaRamp {
-                red:    slice::from_raw_buf(&(llramp.red as *const c_ushort), llramp.size as usize)
+                red:    Vec::from_raw_parts(llramp.red as *mut c_ushort, llramp.size as usize, llramp.size as usize)
```

Comment:
wouldn't this double-allocate? first, when you call `Vec::from_raw_parts`, second when you call `collect()`